### ISSUE TITLE
No Compatible Keyboard found | MSI Katana GF66 11UE

### ISSUE THREAD

**Issue by marioallaa** [Gibtnix/MSIKLM#50]
Is there no support for MSI Katana GF66 11UE, or did I do something wrong whilst installing it? 

If there is no support, do you have any plans on adding it?

**Comment by Gibtnix** [comment]
Did you try `sudo msiklm test` and what is the output? If your keyboard is not found, you can check the device ID of your keyboard using `sudo msiklm list` and check the console output. If your keyboard is listed with a different ID, you can change the ID in `msiklm.c` to use your ID instead. However, the commands most likely won't work as I expect your keyboard to use a different command structure then.
It is possible to extend the support for different devices / keyboards. still this requires the correct command structure.

**Comment by marioallaa** [comment]
`sudo msiklm test`
result: 
```
No compatible keyboard found!
Check you're using sudo!
```

`sudo msikl list`
result
```
Device: BenQ ZOWIE Gaming Mouse
    Device Vendor ID:        1189
    Device Product ID:       32769
    Device Serial Number:    (null)
    Device Manufacturer:     BenQ ZOWIE
    Device Path:             3-8:1.0
    Device Interface Number: 0
    Device Release Number:   32
```

**Comment by marioallaa** [comment]
How do i change the id in msiklm.c, I don't know C, and I am afraid I will ruin something.

**Comment by Gibtnix** [comment]
The output of `sudo msiklm list` is incomplete, I guess you only copied the last block. There should be multiple outputs with the same structure. One of them should be your keyboard, but the device you copied obviously is a gaming mouse.

To open your keyboard, the vendor and product IDs have to be used. However, most likely the command structure will be different such that your keyboard won't work. Still, it might be worth a try. I can tell you how to modify the function and recompile msiklm, still I need the respective IDs for this.

**Comment by marioallaa** [comment]
That is all, if i remove the mouse, this is the output: 

```
No HID device found!
```

**Comment by marioallaa** [comment]
this amazon post says that it does have RGB backlight, but my backlight is only red. 
https://www.amazon.com/MSI-Katana-i7-11800H-Processor-Keyboard/dp/B09TCFVGJF?th=1

**Comment by AtomicRobotMan0101** [comment]
According to the specs, the keyboard is only backlit in red.

https://au.msi.com/Laptop/Katana-GF66-11UX/Specification

One can use the Fn +- to change the brightness, but thats it.

**Comment by Gibtnix** [comment]
If the keyboard is not detect by `msiklm list`, it won't be supported. Most likely it won't be detected by the SteelSeries Engine on Windows, too.

**Comment by varshneydevansh** [comment]
![image](https://github.com/Gibtnix/MSIKLM/assets/20411128/e6d3bdde-856b-4141-8490-ca56ef4276ef)


My device is MSI GP65 Leopard

**Comment by Gibtnix** [comment]
You could try replacing the IDs inside `msiklm.c`, i.e. replace the line
```dev = hid_open(0x1770, 0xff00, 0);```
by
```dev = hid_open(4152, 4386, 0);```
and recompile. Thereafter, try `msiklm test`. It should find your keyboard, however it is unclear whether changing any configuration setting will work or not.

**Comment by sempervictus** [comment]
Seeing the same thing:
```
msiklm list
Device: SteelSeries KLC
    Device Vendor ID:        4152
    Device Product ID:       4386
    Device Serial Number:    (null)
    Device Manufacturer:     SteelSeries
    Device Path:             1-2:1.0
    Device Interface Number: 0
    Device Release Number:   598

Device: SteelSeries KLC
    Device Vendor ID:        4152
    Device Product ID:       4386
    Device Serial Number:    (null)
    Device Manufacturer:     SteelSeries
    Device Path:             1-2:1.1
    Device Interface Number: 1
    Device Release Number:   598

Device: SteelSeries ALC
    Device Vendor ID:        4152
    Device Product ID:       4443
    Device Serial Number:    (null)
    Device Manufacturer:     SteelSeries
    Device Path:             1-11:1.0
    Device Interface Number: 0
    Device Release Number:   579

Device: SteelSeries ALC
    Device Vendor ID:        4152
    Device Product ID:       4443
    Device Serial Number:    (null)
    Device Manufacturer:     SteelSeries
    Device Path:             1-11:1.1
    Device Interface Number: 1
    Device Release Number:   579

```
and while that change does detect it, the keyboard does not illuminate from the fn+f8 combination nor `sudo ./msiklm red high`
What's interesting is that my original titan18hx worked fine (manufacture date dec of 23) but the 14th gen in it went sour and the new one (mfg date of aug 24) does not even if running the original chassis' keyboard... they changed something on the motherboard/EC which is not updated by the .116 bios+ec image.

**Comment by Gibtnix** [comment]
Yes, I agree with that. Still, if there is any information about the new command structure available, it is possible to extend the program. However, someone has to find out what's required for the respective keyboard.

**Comment by sempervictus** [comment]
Is this something i can sniff from the IOs of the windows app if i boot into that?

**Comment by Gibtnix** [comment]
Generally, I think this is possible. However, I don't know which tools are best suited for that. If you find a working command structure by sniffing the traffic running Windows, you can implement the same data (hard code for testing) into one of the functions in `msiklm.c` and try. If you can find out the command structure, it is generally possible to extend msiklm accordingly (e.g. try different IDs in `open_keyboard` and save the respective one to configure the commands accordingly).
